schema.go

```diff
@@ -498,6 +506,44 @@ func (s *RecordSchema) String() string {
 	return `{"name":"` + s.FullName() + `","type":"` + typ + `","fields":[` + fields + `]}`
 }
 
+// Resolve returns the resolved form of the schema.
+func (s *RecordSchema) Resolve(cache *SchemaCache) string {
```

Comment:
You have an issue here with recursive schemas that you are not dealing with. To be clear, `RefSchema`s exist to break this potential cycle.